libs/wallet/provider/uphold/uphold.go

```diff
@@ -843,10 +889,21 @@ func (resp upholdTransactionResponse) ToTransactionInfo() *walletutils.Transacti
 	return &txInfo
 }
 
-// SubmitTransaction submits the base64 encoded transaction for verification but does not move funds
-//
-//	unless confirm is set to true.
+// SubmitTransaction submits the base64 encoded transaction for verification but
+// does not move funds unless confirm is set to true.
 func (w *Wallet) SubmitTransaction(ctx context.Context, transactionB64 string, confirm bool) (*walletutils.TransactionInfo, error) {
+	return w.submitTransaction(ctx, defaultHTTPClient, transactionB64, confirm)
+}
+
+func (w *Wallet) submitTransaction(
```

Comment:
just to clarify, we're reusing the `submitTransaction` call here as a KYC check when confirm is set to false, but when it's set to true we're performing the actual transaction? This isn't clear from reading the code at the moment and this should likely be documented better or the function refactored to split out submitting transactions from KYC checks.

Reply:
The confirm matches Uphold API https://docs.uphold.com/#create-amp-commit-a-transaction . When confirm is false, a transaction is only created and later has to be confirmed using a separated API, ConfirmTransaction. When it is true the transaction is created and confirmed in one go.

The specialty of IsUserKYC is not using `confirm == false` as that is used in other places , but the zero amount. That is commented in `IsUserKYC()`.

Reply:
I will do a separated PR to do an extra cleanup in uphold.go and will clarify in comments the need for the confirm arguments.